File: 백준/Silver/18110. solved．ac/solved．ac.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        if (n == 0) {
            System.out.println(0);
            System.exit(0);
        }

        int[] difficulty = new int[n];
        for (int i = 0; i < n; i++) {
            difficulty[i] = Integer.parseInt(br.readLine());
        }

        double cut = n * 0.15;
        int intCut = (int) Math.round(cut);

        Arrays.sort(difficulty);
        int[] diffCut = new int[(n-(2*intCut))];

        for (int i = intCut; i < n - intCut; i++) {
            diffCut[i-intCut] = difficulty[i];
        }

        double total = 0;
        for (int i = 0; i < diffCut.length; i++) {
            total += diffCut[i];
        }

        total /= diffCut.length;

        System.out.println((int) Math.round(total));
    }
}

File: 백준/Silver/18110. solved．ac/solved．ac.py
#solved.ac
import sys
input = sys.stdin.readline

n = int(input())

if n == 0:
    print(0)
    exit()
    
diffi = []
for i in range(n):
    diffi.append(int(input()))

cut = n * 0.15
if cut * 10 % 10 >= 5:
    cut = int(cut) + 1
else:
    cut = int(cut)
    
diffi.sort()
diffi_cut = diffi[cut:n-cut]

total = 0
for i in diffi_cut:
    total += i

total /= len(diffi_cut)
if total * 10 % 10 >= 5:
    total = int(total) + 1
else:
    total = int(total)

print(total)
